File: Python/Counting_Sort.py
def counting_sort(arr):
    count_map = {}

    for num in arr:
        if num in count_map:
            count_map[num] += 1
        else:
            count_map[num] = 1

    keys = []
    for key in count_map:
        keys.append(key)

    keys.sort()

    j = 0
    for key in keys:
        for _ in range(count_map[key]):
            arr[j] = key
            j += 1

File: C++/Sorting/Counting_Sort.cpp
#include <iostream>
#include <vector>
void Counting_Sort(std::vector<int> arr){

    std::unordered_map<int, int> map;

    for (auto num : arr) {
        map[num]++;
    }

    std::vector<int> keys;
    for (auto num : map) {
        keys.push_back(num.first);
    }

    std::sort(keys.begin(), keys.end());

    int j = 0;
    for (auto key : keys) {
        for (int i = 0; i < map[key]; i++) {
            arr[j++] = key;
        }
    }


}